x-pack/legacy/plugins/siem/public/containers/case/use_delete_cases.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useCallback, useReducer } from 'react';
+import { errorToToaster } from '../../components/ml/api/error_to_toaster';
+import * as i18n from './translations';
+import { useStateToaster } from '../../components/toasters';
+import { deleteCases } from './api';
+
+interface DeleteState {
+  isDisplayConfirmDeleteModal: boolean;
+  isDeleted: boolean;
+  isLoading: boolean;
+  isError: boolean;
+}
+type Action =
+  | { type: 'DISPLAY_MODAL'; payload: boolean }
+  | { type: 'FETCH_INIT' }
+  | { type: 'FETCH_SUCCESS'; payload: boolean }
+  | { type: 'FETCH_FAILURE' }
+  | { type: 'RESET_IS_DELETED' };
+
+const dataFetchReducer = (state: DeleteState, action: Action): DeleteState => {
+  switch (action.type) {
+    case 'DISPLAY_MODAL':
+      return {
+        ...state,
+        isDisplayConfirmDeleteModal: action.payload,
+      };
+    case 'FETCH_INIT':
+      return {
+        ...state,
+        isLoading: true,
+        isError: false,
+      };
+    case 'FETCH_SUCCESS':
+      return {
+        ...state,
+        isLoading: false,
+        isError: false,
+        isDeleted: action.payload,
+      };
+    case 'FETCH_FAILURE':
+      return {
+        ...state,
+        isLoading: false,
+        isError: true,
+      };
+    case 'RESET_IS_DELETED':
+      return {
+        ...state,
+        isDeleted: false,
+      };
+    default:
+      throw new Error();
```

Comment:
Why we need from the reducer to through an error? I think a reducer should do only one thing, get a state and return a new state. Better to return the state without changes.

Reply:
fair enough